src/storage/src/hummock/compactor/mod.rs

```diff
@@ -225,6 +227,49 @@ impl Compactor {
             .await;
         let multi_filter_key_extractor = Arc::new(multi_filter_key_extractor);
 
+        let mut splits: Vec<KeyRange_vec> = vec![];
+        splits.push(KeyRange_vec::new(vec![], vec![]));
+        // collect sstable_ids to get meta data
+
+        let sstable_ids = select_table_infos
+            .iter()
+            .map(|sstable_info| sstable_info.id)
+            .collect_vec();
+        let mut indexes = vec![];
+
+        // preload the meta and get the smallest key to split sub_compaction
+        for sstable_id in sstable_ids {
+            // extend
+            indexes.extend(
+                context
+                    .sstable_store
+                    .sstable(sstable_id, &mut StoreLocalStatistic::default())
+                    .await
+                    .unwrap()
+                    .value()
+                    .meta
+                    .block_metas
+                    .iter()
+                    .map(|block| {
+                        FullKey::from_user_key_slice(user_key(&block.smallest_key), u64::MAX)
+                            .into_inner()
+                    })
+                    .collect_vec(),
+            );
+        }
+        indexes.sort();
+        // every 8 index as one group
+        const SPLIT_RANGE_STEP: usize = 8;
+        let concurrency = indexes.len() / SPLIT_RANGE_STEP;
```

Comment:
Shall we need to limit the max_concurrency for one compactor ?

Such as that the instance of `Compactor Node` has only one core

Reply:
Yes. We need to add some config.

Reply:
We can not set concurrency by SPLIT_RANGE_STEP. We need to set it by the total file size of each range of compaction. We can know the size of each block in block meta